### PR TITLE
solana: Add tilt and docker targets for Anchor testing

### PR DESCRIPTION
**Pull request by johnsaigle**
## Overview

This PR modifies the repo's `Tiltfile`, adding a new docker image and kubernetes resource to assist with integration testing.

Overall goals:
* Adds new Docker container to build anchor programs and copy the compiled files into a location that the Tile pipeline can access
* Add new Kubernetes resource for compiling anchor. Deploy the anchor program to Wormhole's solana-devnet
* Commit compiled IDL to the repo, so it can be combined with existing integration tests

## Status
- Tilt can build the Docker container for Anchor, which in turn compiles the programs without an issue
- Local testing with `solana program deploy ...` against `solana-test-validator` appears to work
- The actual deployment address has not been added
- No TypeScript or integration test have been modified. This just sets up some plumbing

### Outstanding tasks
- [ ] Create a deterministic keypair for the NTT anchor program deployments
- [ ] Modify `anchor-deploy.yaml` to add `--program-id` to the `solana program deploy` command
- [ ] Merge the anchor integration tests with the EVM tests. (Namely: combine `solana/tests/example-native-token-transfer.ts` with `ci_tests/src/index.ts`)
- [ ] Modify anchor TS tests to use `solana-devnet` as the RPC provider
- [ ] Add `tilit ci -- ... --solana` to GitHub Actions

### Possible additional tasks
- [ ] Rebuild and commit the IDL file, after rebasing/merging with other parallel changes (Build the anchor programs, then copy the JSON IDL file into `ci_tests`)
- [ ] Add an appropriate `readinessProbe` to `anchor-deploy.yaml`
- [ ] Verify that `solana/Dockerfile` has copied the `.so` files into a location that is accessible by the relevant pieces of the Tiltfile pipeline

### CODE REVIEW

File: solana/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,68 @@
+# This file is based on: https://github.com/wormhole-foundation/wormhole/blob/wen/solana-rewrite/solana/Dockerfile.ci
+FROM ubuntu:24.04
```
Comment (SEJeff) [inline solana/Dockerfile:2]: I can't believe this tag is out. The release isn't actually until April 25.

File: solana/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+RUN mkdir -p /workdir && \
+    apt-get update -qq && apt-get upgrade -qq && apt-get install -qq \
+    build-essential git curl wget jq pkg-config python3-pip \
+    libssl-dev libudev-dev
+
+#RUN wget http://nz2.archive.ubuntu.com/ubuntu/pool/main/o/openssl/libssl1.1_1.1.1f-1ubuntu2.19_amd64.deb
+#RUN dpkg -i libssl1.1_1.1.1f-1ubuntu2.19_amd64.deb
+
+RUN wget http://nz2.archive.ubuntu.com/ubuntu/pool/main/o/openssl/libssl1.1_1.1.1f-1ubuntu2.21_amd64.deb
+RUN dpkg -i libssl1.1_1.1.1f-1ubuntu2.21_amd64.deb
```
Comment (SEJeff) [inline solana/Dockerfile:30]: Are these haxx required? Will it work to just use the (currently LTS) `ubuntu:22.04` release? I do not love hardcoding the mirror url if it's possible not to. If it is, please add a comment as to why.

File: solana/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+RUN npm install -g yarn
+
+# Install Solana tools.
+RUN sh -c "$(curl -sSfL https://release.solana.com/${SOLANA_CLI}/install)"
+
+# Install anchor.
+RUN cargo install --git https://github.com/coral-xyz/anchor --tag ${ANCHOR_CLI} anchor-cli --locked
+
+# TODO can this be done more efficiently? Also, refer to .dockerignore
```
Comment (SEJeff) [inline solana/Dockerfile:54]: It can only be done more efficiently if you copy specific things here.

File: solana/Makefile [hunk 4]
```diff
@@ -22,3 +25,10 @@ node_modules: package-lock.json
 clean:
 	anchor clean
 	rm -rf .anchor node_modules
+
+docker:
+	docker build -t solana-ntt-tests \
+		-f Dockerfile \
+		--build-arg SOLANA_CLI=$(SOLANA_CLI) \
+		--build-arg ANCHOR_CLI=$(ANCHOR_CLI) \
+		.
```
Comment (SEJeff) [inline solana/Makefile:34]: Will this run at some point via tilt? This part might need to be duplicated in any Tiltfile that builds this as an image.

Comment (johnsaigle) [reply solana/Makefile:34]: The idea is definitely to integrate with Tilt. I added this for convenience but have no opposition to removing it. 
Compare with this Solana `Makefile` in the monorepo. https://github.com/wormhole-foundation/wormhole/blob/00f504ef452ae2d94fa0024c026be2d8cf903ad5/solana/Makefile#L91

This target should be doing the same thing, but could perhaps be merged with the `test` target.

**Comment by evan-gray** [comment]
Superceded by #259

**Pull request closed without merging**